From e884b98b248e155cda4c53a302cc821a3c2ddaf9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 8 May 2015 15:49:08 +0100
Subject: [PATCH] misspell some words, copy and paste slide to gimp

and we don't want to see the mis-spelt underlines

Change-Id: Ib3ae29297a1697e0f7b0caa816504da7edf721d1
---
 sd/source/ui/app/sdxfer.cxx | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/sd/source/ui/app/sdxfer.cxx b/sd/source/ui/app/sdxfer.cxx
index dda4f74..4ac1fac 100644
--- a/sd/source/ui/app/sdxfer.cxx
+++ b/sd/source/ui/app/sdxfer.cxx
@@ -518,13 +518,27 @@ bool SdTransferable::GetData( const DataFlavor& rFlavor, const OUString& rDestDo
         }
         else if( nFormat == FORMAT_GDIMETAFILE )
         {
-            if( mpSdViewIntern )
+            if (mpSdViewIntern)
+            {
+                const bool bToggleOnlineSpell = mpSdDrawDocumentIntern && mpSdDrawDocumentIntern->GetOnlineSpell();
+                if (bToggleOnlineSpell)
+                    mpSdDrawDocumentIntern->SetOnlineSpell(false);
                 bOK = SetGDIMetaFile( mpSdViewIntern->GetMarkedObjMetaFile( true ), rFlavor );
+                if (bToggleOnlineSpell)
+                    mpSdDrawDocumentIntern->SetOnlineSpell(true);
+            }
         }
         else if( FORMAT_BITMAP == nFormat || SOT_FORMATSTR_ID_PNG == nFormat )
         {
-            if( mpSdViewIntern )
+            if (mpSdViewIntern)
+            {
+                const bool bToggleOnlineSpell = mpSdDrawDocumentIntern && mpSdDrawDocumentIntern->GetOnlineSpell();
+                if (bToggleOnlineSpell)
+                    mpSdDrawDocumentIntern->SetOnlineSpell(false);
                 bOK = SetBitmapEx( mpSdViewIntern->GetMarkedObjBitmapEx(true), rFlavor );
+                if (bToggleOnlineSpell)
+                    mpSdDrawDocumentIntern->SetOnlineSpell(true);
+            }
         }
         else if( ( nFormat == FORMAT_STRING ) && mpBookmark )
         {
-- 
2.1.0

